client/lib/purchases/index.js

```diff
@@ -347,15 +351,16 @@ function isPaidWithCreditCard( purchase ) {
 }
 
 function isPaidWithPayPalDirect( purchase ) {
-	return 'paypal_direct' === purchase.payment.type && purchase.payment.expiryMoment;
+	return 'paypal_direct' === purchase.payment.type && purchase.payment.expiryDate;
 }
 
 function hasCreditCardData( purchase ) {
-	return Boolean( purchase.payment.creditCard.expiryMoment );
+	return Boolean( purchase.payment.creditCard.expiryDate );
 }
 
-function shouldAddPaymentSourceInsteadOfRenewingNow( expiryMoment ) {
-	return expiryMoment > moment().add( 3, 'months' );
+function shouldAddPaymentSourceInsteadOfRenewingNow( purchase ) {
+	const expiry = purchase.expiryDate ? moment( purchase.expiryDate ) : null;
+	return expiry > moment().add( 3, 'months' );
```

Comment:
This is another place that should be more explicit about what's the return value when `purchase.expiryDate` is empty.

Reply:
Indeed. Fixed!